src/lib/Libwin/passwd.c

```diff
@@ -4145,9 +4146,13 @@ wsystem(char *cmdline, HANDLE user_handle)
 			NULL, NULL, TRUE, flags,
 			NULL, NULL, &si, &pi);
 	} else {
+		flags = flags|CREATE_UNICODE_ENVIRONMENT;
+		if (!CreateEnvironmentBlock(&user_env, user_handle, FALSE)) {
+			log_err(-1, __func__, "failed in CreateEnvironmentBlock");
```

Comment:
if we fail to create the environment block, shouldn't we return out of the function? Instead of continuing to run the process.

Reply:
sudehsna addressed it, hence resolving this